packages/tasks_panel/src/misc/cap-e2e-config.ts

```diff
@@ -0,0 +1,102 @@
+import { RelativePattern, Uri, commands } from "vscode";
+import { ChildProcessWithoutNullStreams, spawn } from "child_process";
+import { getLogger } from "../../src/logger/logger-wrapper";
+import {
+  CAP_DEPLOYMENT_CONFIG,
+  LabelType,
+  ProjectInfo,
+  ProjectTypes,
+  addTaskDefinition,
+  areResourcesReady,
+  generateMtaDeployTasks,
+  isFileExist,
+  isTasksSettled,
+  waitForResource,
+} from "./e2e-config";
+import { last } from "lodash";
+
+export interface CapProjectConfigInfo extends ProjectInfo {
+  type: string;
+}
+
+/* istanbul ignore next */
+function invokeCommand(config: { cmd: string; args: string[]; cwd: string }, additionalArgs: string[]): Promise<void> {
+  function executeSpawn(
+    config: { cmd: string; args: string[]; cwd: string },
+    additionalArgs: string[]
+  ): ChildProcessWithoutNullStreams {
+    const { args, cmd, cwd } = config;
+    return spawn(cmd, [...args, ...additionalArgs], { cwd });
```

Comment:
I've had some success with using [execa](https://github.com/sindresorhus/execa)
Instead of `child_process` directly.

Would it be useful for your usecase as well?